Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/ViewBookmarkLoader.cpp

```diff
@@ -0,0 +1,398 @@
+                            m_localBookmarks = viewBookmarkVisitor.m_bookmarkMap.at(m_bookmarkfileName);
+                            m_localBookmarkCount = m_localBookmarks.size() - 1;
+                            m_lastKnownLocation = viewBookmarkVisitor.m_lastKnownLocation;
+                        }
+
+                        // once loaded we can remove the data from the settings registry.
+                        registry->Remove(bookmarkKey + "/");
+                        return visitedViewBookmarks;
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        ViewBookmark ViewBookmarkLoader::GetBookmarkAtIndex(int index) const
+        {
+            ViewBookmarkComponent* bookmarkComponent = FindBookmarkComponent();
+            if (bookmarkComponent)
+            {
+                return bookmarkComponent->GetBookmarkAtIndex(index);
+            }
+
+            return ViewBookmark();
+        }
+
+        ViewBookmark ViewBookmarkLoader::GetLastKnownLocationInLevel() const
+        {
+            return ViewBookmark();
+        }
+
+        ViewBookmarkComponent* ViewBookmarkLoader::FindBookmarkComponent() const
+        {
+            AZ::EntityId levelEntityId;
+            AzToolsFramework::ToolsApplicationRequestBus::BroadcastResult(
+                levelEntityId, &AzToolsFramework::ToolsApplicationRequests::GetCurrentLevelEntityId);
+
+            if (levelEntityId.IsValid())
+            {
+                AZ::Entity* levelEntity = nullptr;
+                AZ::ComponentApplicationBus::BroadcastResult(levelEntity, &AZ::ComponentApplicationBus::Events::FindEntity, levelEntityId);
+                if (levelEntity)
+                {
+                    ViewBookmarkComponent* bookmarkComponent = levelEntity->FindComponent<ViewBookmarkComponent>();
+                    if (bookmarkComponent)
+                    {
+                        return bookmarkComponent;
+                    }
+                }
+            }
+            return nullptr;
+        }
+
+        AZStd::string ViewBookmarkLoader::GenerateBookmarkFileName() const
+        {
+            AZ::EntityId levelEntityId;
+            AzToolsFramework::ToolsApplicationRequestBus::BroadcastResult(
+                levelEntityId, &AzToolsFramework::ToolsApplicationRequests::GetCurrentLevelEntityId);
+
+            if (levelEntityId.IsValid())
+            {
+                AZ::Entity* levelEntity = nullptr;
+                AZ::ComponentApplicationBus::BroadcastResult(levelEntity, &AZ::ComponentApplicationBus::Events::FindEntity, levelEntityId);
+                if (levelEntity)
+                {
+                    AZStd::chrono::system_clock::time_point now = AZStd::chrono::system_clock::now();
+                    AZStd::string sTime = AZStd::string::format("%llu", now.time_since_epoch().count());
+
+                    return levelEntity->GetName() + sTime + ".setreg";
+                }
```

Comment:
suggestion: A comment for this might be nice, and also a cool way to test this could be to provide a callback for something like `GenerateUniqueId` that returns the time in reality but from a can be stubbed in as just a number we know and can verify. That makes testing for potential collisions/duplicates easier too